### PR TITLE
Add pilot_sds_certificate_errors_total metric for sds

### PR DESCRIPTION
**Pull request by kebe7jun**
For https://github.com/istio/istio/issues/31779

[X] Policies and Telemetry

### CODE REVIEW

File: pilot/pkg/xds/monitoring.go [hunk 1]
```diff
@@ -167,6 +167,12 @@ var (
 		monitoring.WithLabels(typeTag),
 	)
 
+	proxySDSCertificateErrors = monitoring.NewGauge(
+		"proxy_sds_certificate_errors",
+		"Number of failed to fetch key and certificate.",
+		monitoring.WithLabels(nodeTag),
```
Comment (bianpengyuan) [inline pilot/pkg/xds/monitoring.go:173]: Please do not use unbounded value as metric label.

Comment (kebe7jun) [reply pilot/pkg/xds/monitoring.go:173]: I saw https://github.com/istio/istio/blob/3b9c6b88e0696577f4f1d0e93d7c28912828790d/pilot/pkg/xds/monitoring.go#L59 is also used like this.

Is there any suggestions?
Use secret name and ns as label?

Comment (bianpengyuan) [reply pilot/pkg/xds/monitoring.go:173]: Looks like secret name and namespace has already be logged. IMO that should be good enough, and this metric does not necessarily need to have any label. Alert can be built with this metric and user could check istiod logs to troubleshoot.

Also metric name probably should be `pilot_sds_certificate_errors_total` for consistency with exisiting metric as well as prometheus recommendation: https://prometheus.io/docs/practices/naming/

File: releasenotes/notes/31779.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: telemetry
```
Comment (bianpengyuan) [inline releasenotes/notes/31779.yaml:3]: nit: we typically categorize new control plane stats as network feature.

File: pilot/pkg/xds/monitoring.go (hunk 1 above)
Comment (kebe7jun) [reply pilot/pkg/xds/monitoring.go]: thanks very much!

**Review by bianpengyuan — COMMENTED** [review]
LGTM. Will let folks with more knowledge about sds code path to take a look. @howardjohn @JimmyCYJ

File: pilot/pkg/xds/monitoring.go [hunk 3]
```diff
@@ -167,6 +167,11 @@ var (
 		monitoring.WithLabels(typeTag),
 	)
 
+	pilotSDSCertificateErrors = monitoring.NewSum(
+		"pilot_sds_certificate_errors_total",
+		"Total number of failed to fetch key and certificate.",
```
Comment (bianpengyuan) [inline pilot/pkg/xds/monitoring.go:172]:
```suggestion
		"Total number of failures to fetch SDS key and certificate.",
```

**Review by howardjohn — APPROVED** [review]

**Pull request merged**